publishing/s3publisher.py

```diff
@@ -193,7 +193,8 @@ def publish_to_s3(directory, base_url, site_prefix, bucket, cache_control,
         if not local_objects_by_filename.get(filename)
     ]
 
-    if len(new_objects) == 0 and len(replacement_objects) == 0:
+    if (len(new_objects) == 0 and len(replacement_objects) <= 1 and
+            len(local_files) <= 1):
```

Comment:
we could probably just use `len(local_files) <= 1`...

Reply:
agreed ... however, its a problem for our tests :(

Reply:
Ideally, our tests would reflect what we want the logic to be, right now we're not testing this at all, maybe we should?